tests/std/tests/P0784R7_library_support_for_more_constexpr_containers/test.cpp

```diff
@@ -272,8 +276,152 @@ constexpr void test_compiletime() {
     }
 }
 static_assert((test_compiletime(), true));
+
+template <class T>
+struct A {
+    T value;
+
+    constexpr A() noexcept = default;
+    constexpr ~A()         = default;
+};
+
+template <class T>
+struct nontrivial_A {
+    T value;
+
+    constexpr nontrivial_A(T in = T{}) noexcept : value(in) {}
+    constexpr ~nontrivial_A() {}
+};
+
+#if defined(__clang__) || defined(__EDG__)
+template <class T, bool Construct = false, bool Destroy = false>
+struct Alloc {
+    using value_type = T;
+    using size_type  = std::size_t;
+
+    constexpr Alloc(int id_) noexcept : id(id_) {}
+
+    constexpr value_type* allocate(size_t n) {
+        assert(n == 10);
+        return allocator<T>{}.allocate(n);
+    }
+
+    constexpr void deallocate(value_type* ptr, size_t n) {
+        assert(n == 10);
+        allocator<T>{}.deallocate(ptr, n);
+    }
+
+    constexpr void construct(value_type* ptr, value_type n) requires(Construct) {
+        allocator<T>{}.construct(ptr, n);
+    }
+
+    constexpr void destroy(value_type* ptr) requires(Destroy) {
+        allocator<T>{}.destroy(ptr);
+    }

```

Comment:
I am not sure whether this is correct, we never really remove something so we should test for someone pulling the escape hatch too

Reply:
Pulling the escape hatch as in `_Alloc.construct` / `_Alloc.destroy` not being available, and needing to call `construct_at` and `destroy_at` instead?  I believe the `requires` should conditionally remove `construct` or `destroy` from `Alloc`'s member functions, so that is how we are exercising this escape hatch.  

Getting into the details of the escape hatch and corresponding changes: 
After consulting with @StephanTLavavej and @CaseyCarter, I don't think we have a nice way to verify that we are calling `std::construct_at` / `std::destroy_at` instead of `::new` and `~_Ty()` (one of the changes in WG21-P0784).  Just seeing the change in the product code is sufficient and consistent with similar changes elsewhere, at least for the time being.